code/game/objects/structures/crates_lockers/closets/secure/freezer.dm

```diff
@@ -33,33 +35,18 @@
 	flags_1 &= ~PREVENT_CONTENTS_EXPLOSION_1
 	return FALSE
 
-/obj/structure/closet/secure_closet/freezer/atom_destruction(damage_flag)
-	new /obj/item/stack/sheet/iron(drop_location(), 1)
-	new /obj/item/assembly/igniter/condenser(drop_location())
-	return ..()
-
-/obj/structure/closet/secure_closet/freezer/welder_act(mob/living/user, obj/item/tool)
+/obj/structure/closet/secure_closet/freezer/deconstruct(disassembled)
+	if (!(flags_1 & NODECONSTRUCT_1))
+		new /obj/item/assembly/igniter/condenser(drop_location())
 	. = ..()
 
-	if(!opened)
-		balloon_alert(user, "open it first!")
-		return TRUE
-
-	if(!tool.use_tool(src, user, 40, volume=50))
-		return TRUE
-
-	new /obj/item/stack/sheet/iron(drop_location(), 2)
-	new /obj/item/assembly/igniter/condenser(drop_location())
-	qdel(src)
-
-	return TRUE
-
 /obj/structure/closet/secure_closet/freezer/empty
 	name = "freezer"
 
 /obj/structure/closet/secure_closet/freezer/empty/open
 	req_access = null
 	locked = FALSE
+	opened = 1
```

Comment:
```suggestion
	opened = TRUE
```

And don't forget the open icon state

Reply:
Yup

Reply:
Closets don't have a proper opened icon state. the door is only added as an overlay
![Screenshot (199)](https://user-images.githubusercontent.com/110812394/236463418-a296a8a4-5cfe-4326-bdf7-81884e7e87af.png)